File: ruby/src/Behavioral/Chain_Of_Responsibility/DvdCategory.rb
require "Behavioral/Chain_Of_Responsibility/TopTitle"

class DvdCategory < TopTitle


	def initialize(category)
 		setCategory(category)
		@topCategoryTitle = nil
	end

	def setCategory(categoryIn)
 		@category = categoryIn
	end

	def getCategory()
 		return  @category
	end

	def getAllCategories()
 		return getCategory()
	end

	def setTopCategoryTitle(topCategoryTitleIn)
 		@topCategoryTitle = topCategoryTitleIn
	end

	def getTopCategoryTitle()
 		return @topCategoryTitle
	end

	def getTopTitle()
 		return @topCategoryTitle
	end
end

File: csharp/src/Behavioral/Chain_Of_Responsibility/DvdCategory.cs
//DvdCategory.cs - the Highest Class in the Chain

namespace Behavioral.Chain_Of_Responsibility {

using System;

public class DvdCategory : TopTitle {  
   private string category; 
   private string topCategoryTitle;
    
   public DvdCategory(String category) {
        this.setCategory(category);
   }    
   
   public void setCategory(String categoryIn) {
       this.category = categoryIn;
   }
   public string getCategory() {
       return this.category;
   } 
   public string getAllCategories() {
       return getCategory();
   }
   
   public void setTopCategoryTitle(String topCategoryTitleIn) {
       this.topCategoryTitle = topCategoryTitleIn;
   }
   public string getTopCategoryTitle() {
       return this.topCategoryTitle;
   }
   
   public string getTopTitle() {
       return this.topCategoryTitle;
   }
}

}
